lms/djangoapps/commerce/api/v0/views.py

```diff
@@ -121,26 +124,49 @@ def post(self, request, *args, **kwargs):
         if not default_enrollment_mode:
             msg = Messages.NO_DEFAULT_ENROLLMENT_MODE.format(course_id=course_id)
             return DetailResponse(msg, status=HTTP_406_NOT_ACCEPTABLE)
-        elif default_enrollment_mode and not default_enrollment_mode.sku:
-            # If there are no course modes with SKUs, enroll the user without contacting the external API.
-            msg = Messages.NO_SKU_ENROLLED.format(
-                enrollment_mode=default_enrollment_mode.slug,
-                course_id=course_id,
-                username=user.username
+        elif not default_enrollment_mode.sku or STOP_BASKET_CREATION_FLAG.is_enabled():
+            msg = Messages.ENROLL_DIRECTLY.format(
+                username=user.username,
+                course_id=course_id
             )
+            if not default_enrollment_mode.sku:
+                # If there are no course modes with SKUs, return a different message.
+                msg = Messages.NO_SKU_ENROLLED.format(
+                    enrollment_mode=default_enrollment_mode.slug,
+                    course_id=course_id,
+                    username=user.username
+                )
             log.info(msg)
             self._enroll(course_key, user, default_enrollment_mode.slug)
             self._handle_marketing_opt_in(request, course_key, user)
             return DetailResponse(msg)
+        else:
+            return self._create_basket_to_order(request, user, course_key, default_enrollment_mode)
 
-        # Setup the API
+    def _add_request_cookie_to_api_session(self, server_session, request, cookie_name):
```

Comment:
The method was not moved. Instead I moved a big block of code right above this function to a new function underneath. I think that tricked github compare tool